lib/classes/class-errors.php

```diff
@@ -131,8 +132,12 @@ public function has_errors() {
        * @since 2.1.0
        */
       public function dismiss() {
-        if ( isset( $_GET[ 'udan-dismiss-' . sanitize_key( $this->name ) ] ) ) {
+        if ( isset( $_GET[ 'udan-dismiss-notice' . sanitize_key( $this->name ) ] ) ) {
           update_option( ( 'dismissed_notice_' . sanitize_key( $this->name ) ), time() );
+        } else if ( isset( $_GET[ 'udan-dismiss-warning' . sanitize_key( $this->name ) ] ) ) {
+          update_option( ( 'dismissed_warning_' . sanitize_key( $this->name ) ), time() );
+        } else if ( isset( $_GET[ 'udan-dismiss-error' . sanitize_key( $this->name ) ] ) ) {
+          update_option( ( 'dismissed_error_' . sanitize_key( $this->name ) ), time() );
         }
```

Comment:
For all cases (notices, errors) you are using the same option. I guess, it breaks logic. In case, you will dismiss notice, warning will be dismissed too on page reload. Am I wrong?